### PR TITLE
[SPARK-4989][CORE] avoid wrong eventlog conf cause cluster down in standalone mode

### PR DESCRIPTION
**Pull request by liyezhang556520**
when enabling eventlog in standalone mode, if give the wrong configuration, the standalone cluster will down (cause master restart, lose connection with workers).
How to reproduce: just give an invalid value to "spark.eventLog.dir", for example: spark.eventLog.dir=hdfs://tmp/logdir1, hdfs://tmp/logdir2. This will throw illegalArgumentException, which will cause the Master restart. And the whole cluster is not available.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/master/Master.scala [hunk 1]
```diff
@@ -758,13 +760,14 @@ private[spark] class Master(
         // Event logging is enabled for this application, but no event logs are found
         val title = s"Application history not found (${app.id})"
         var msg = s"No event logs found for application $appName in $eventLogFile."
-        logWarning(msg)
+        val exception = URLEncoder.encode(Utils.exceptionString(fnf), "UTF-8")
+        logWarning(msg, fnf)
         msg += " Did you specify the correct logging directory?"
         msg = URLEncoder.encode(msg, "UTF-8")
-        app.desc.appUiUrl = notFoundBasePath + s"?msg=$msg&title=$title"
+        app.desc.appUiUrl = notFoundBasePath + s"?msg=$msg&exception=$exception&title=$title"
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/master/Master.scala]: I think it's OK to not encode the exception here. This is a file not found exception so it doesn't convey much extra information.

File: core/src/main/scala/org/apache/spark/deploy/master/Master.scala [hunk 2]
```diff
@@ -757,11 +758,12 @@ private[spark] class Master(
       case fnf: FileNotFoundException =>
         // Event logging is enabled for this application, but no event logs are found
         val title = s"Application history not found (${app.id})"
-        var msg = s"No event logs found for application $appName in $eventLogFile."
-        logWarning(msg)
+        var msg = s"No event logs found for application $appName in ${app.desc.eventLogDir}."
+        val exception = URLEncoder.encode(Utils.exceptionString(fnf), "UTF-8")
+        logWarning(msg, fnf)
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/master/Master.scala]: as commented elsewhere, I don't think logging the stack trace here is particularly useful

File: core/src/main/scala/org/apache/spark/deploy/master/Master.scala (hunk 1 above)
Comment (liyezhang556520) [inline core/src/main/scala/org/apache/spark/deploy/master/Master.scala]: ok, I'll revert the change, thanks @andrewor14 , @JoshRosen , @srowen

File: core/src/main/scala/org/apache/spark/deploy/master/Master.scala [hunk 3]
```diff
@@ -757,7 +758,7 @@ private[spark] class Master(
       case fnf: FileNotFoundException =>
         // Event logging is enabled for this application, but no event logs are found
         val title = s"Application history not found (${app.id})"
-        var msg = s"No event logs found for application $appName in $eventLogFile."
+        var msg = s"No event logs found for application $appName in ${app.desc.eventLogDir}."
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/master/Master.scala]: great. This used to say something like "No event logs found for application X in /history/not-found"

**Comment by andrewor14** [comment]
Ok LGTM I'm merging this into master thanks

**Comment by andrewor14** [comment]
Hey @liyezhang556520 this doesn't merge cleanly in older branches. Can you make new PRs against those branches?

**Comment by liyezhang556520** [comment]
@andrewor14 , I received an email of your comment about creating other PRs to fix this issue for other older branches, but not found on this page. I think you might have removed that comment, so do I still need to make new PRs or just ignore that message?

**Comment by andrewor14** [comment]
Yes that would be great. It seems that not all of the changes in this PR are applicable there, however.

**Comment by liyezhang556520** [comment]
ok, I'll make new PRs for those old branches 1.0, 1.1, and 1.2.

**Pull request closed without merging**